src/odoc/odoc_link.ml

```diff
@@ -41,9 +41,10 @@ let from_odoc ~resolver ~warnings_options input output =
       >>= fun (page, warnings) ->
       Odoc_file.save_page output ~warnings page;
       Ok (`Page page)
-  | Unit_content (m, shape) ->
-      link_unit ~resolver ~filename m shape
+  | Unit_content m ->
+      link_unit ~resolver ~filename m
       |> handle_warnings ~input_warnings ~warnings_options
       >>= fun (m, warnings) ->
-      Odoc_file.save_unit output ~warnings (m, shape);
+      let m = {m with Odoc_model.Lang.Compilation_unit.shape = None} in
```

Comment:
Doesn't it make more sense to modify the compilation unit record only in the `link_unit` function?
When a record is modified in many place, I find it difficult to have an idea of which field will go into the file...

Moreover, it would be nice to add a small comment, just in case it is not clear later (`shapes are not used after the link phase`).